nautobot/ipam/views.py

```diff
@@ -804,6 +819,159 @@ def post(self, request):
+            messages.warning(request, msg)
+            return redirect(self.get_return_url(request))
+
+    def get(self, request):
+        return self.find_duplicate_ips(request)
+
+    def post(self, request):
+        logger = logging.getLogger(__name__)
+        collapsed_ips = IPAddress.objects.filter(pk__in=request.POST.getlist("pk"))
+        merged_attributes = request.POST
+        operation_invalid = len(collapsed_ips) < 2
+        # Check if there are at least two IP addresses for us to merge
+        # and if the skip button is pressed instead.
+        if "_skip" not in request.POST and not operation_invalid:
+            with cache.lock("ipaddress_merge", blocking_timeout=15, timeout=settings.REDIS_LOCK_TIMEOUT):
+                with transaction.atomic():
+                    namespace = Namespace.objects.get(pk=merged_attributes.get("namespace"))
+                    status = Status.objects.get(pk=merged_attributes.get("status"))
+                    # Retrieve all attributes from the request.
+                    if merged_attributes.get("tenant"):
+                        tenant = Tenant.objects.get(pk=merged_attributes.get("tenant"))
+                    else:
+                        tenant = None
+                    if merged_attributes.get("role"):
+                        role = Role.objects.get(pk=merged_attributes.get("role"))
+                    else:
+                        role = None
+                    if merged_attributes.get("tags"):
+                        tag_pk_list = merged_attributes.get("tags").split(",")
+                        tags = Tag.objects.filter(pk__in=tag_pk_list)
+                    else:
+                        tags = []
+                    if merged_attributes.get("nat_inside"):
+                        nat_inside = IPAddress.objects.get(pk=merged_attributes.get("nat_inside"))
+                    else:
+                        nat_inside = None
+                    # use IP in the same namespace as a reference.
+                    ip_in_the_same_namespace = collapsed_ips.filter(parent__namespace=namespace).first()
+                    merged_ip = IPAddress(
+                        host=merged_attributes.get("host"),
+                        ip_version=ip_in_the_same_namespace.ip_version,
+                        parent=ip_in_the_same_namespace.parent,
+                        type=merged_attributes.get("type"),
+                        status=status,
+                        role=role,
+                        dns_name=merged_attributes.get("dns_name", ""),
+                        description=merged_attributes.get("description"),
+                        mask_length=merged_attributes.get("mask_length"),
+                        tenant=tenant,
+                        nat_inside=nat_inside,
+                        _custom_field_data=ip_in_the_same_namespace._custom_field_data,
+                    )
+                    merged_ip.tags.set(tags)
```

Comment:
Can you safely do this before you call `merged_ip.save()` for the first time? TIL.

Reply:
It seems like it works as expected after some manual testing and unit tests that are testing `tags` specifically are passing as well.